2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,210 @@
+For another training session the lead trainer would just have to fill the web-form again, overwriting previous values.
+![HTML MOCKUP ALL_APPLICATIONS]( https://cloud.githubusercontent.com/assets/15982349/13816688/66a3e7e4-ebb5-11e5-83f1-5086ddddac13.png)
+
+3. All accepted applications, are then transferred to the Trainee view. In this view the lead trainer, trainers,discussion leaders, lesson maintainers can track progress of the trainees, provide feedback and update the status of every stage. They are also notified here if the Trainee has requested a discussion session.
+This list can be filtered on the following criteria:
+  * instructor assigned.
+  * start date of training.
+  * location of the training.
+
+  When the training is completed and the trainee has been certified, the lead instructor can export the trainee as instructor, to move their accounts to the Person table, and remove them from the Trainee table.
+![HTML MOCKUP ALL_TRAINEE](https://cloud.githubusercontent.com/assets/15982349/13816909/3401c4a4-ebb6-11e5-9799-0eb21ccd5eea.png)
+4. *Trainee Home page* -
+After getting accepted for the training program, the trainees recieve a randomly generated password, which they can use to log in to AMY, a trainee's home page is different from the AMY DASHBOARD, the AMY DASHBOARD is reservered for administrative purposes only. After logging in the trainees are directed to TRAINEE_HOME. This view will have a nav-bar - **HOME** , **ABOUT** and **NOTIFICATION**.
+  * *HOME*  shows the trainees progress, through a progress bar indicating which stages have been completed and the ones left. In this view the trainee can also request a discussion session by filling in a webform, with fields such as TOPIC, PREFERRED TIME, and so on.
+This view will also list down the general guidelines for instructor training, informing what is required in each stage of the training.
+  * *ABOUT* , this view shows the information entered by the trainee in their initial application, they can also edit this information, and change passwords here.
+  * *NOTIFICATION*,  this view allows the trainee to view all the comments/feedback from the discussion leaders and the lesson maintainers.In the format *comment: by whom: timestamp*.
+
+
+## Technical Details
+The Software stack for this project can be divided into:
+
+ * Backend  : [Django](https://www.djangoproject.com/)
+ * Database : [SQLite](https://www.sqlite.org/)
+ * Frontend : [Bootstrap](http://getbootstrap.com/),[JQuery](https://jquery.com/) for AJAX requests
+
+### Features and Model Schemes
+
+Trainee would be the main Model, maintaining most of the Flags for Progress and other information:
+
+```python
+lass Trainee(AbstractBaseUser):
+    '''Model for a Trainee '''
+    # listing just essential fields, more fields can and should be added when implementing
+    COURSE_CHOICES = (
+        ('1', 'SWC'),
+        ('2', 'DC'),
+        )
+    personal    = models.CharField(max_length=STR_LONG)
```

Comment:
1. :-1: I strongly recommend using Person for every person we track, and having each Trainee record link to a Person.
2. `course` should not be a text field - what do you think it should be instead? (Hint: a foreign key, but to what?)
3. What are the purposes of the three Boolean fields? What is the state if someone hasn't yet submitted homework or taken part in a discussion?